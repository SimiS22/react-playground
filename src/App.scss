@import "~antd/dist/antd.css";
.app {
  width: 100%;
  height: 100%;
  margin: 0;
  padding: 0;
  .content-area {
    display: flex;
  }
}
