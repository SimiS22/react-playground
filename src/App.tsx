import React, { useState } from 'react';
import './App.scss';
import Header from './components/Header/Header';
import Sidebar from './components/Sidebar/Sidebar';
import Content from './pages/Content/Content';

export const TestApp: React.FC = () => {
  const [text, setText] = useState('initial value');
  return (
    <>
      <div data-testid="myDiv">{text}</div>
      <button data-testid="myButton" onClick={() => setText('final value')}>Change</button>
    </>
  );
};












const App: React.FC = () => {
  return (
    <div className="app">
      <Header />
      <div className="content-area">
        <Sidebar />
        <Content />
      </div>
    </div>
  );
};
export default App;
