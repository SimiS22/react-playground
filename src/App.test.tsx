import { fireEvent, render } from '@testing-library/react';
import React from 'react';
import { TestApp } from './App';

describe('test', () => {
  it('test 1', () => {
    const rendered = render(<TestApp />);

    const myDiv = rendered.getByTestId('myDiv');
    const myButton = rendered.getByTestId('myButton');

    expect(myDiv).toHaveTextContent('initial value');

    fireEvent.click(myButton);

    expect(myDiv).toHaveTextContent('final value');

    expect(rendered.container).toMatchSnapshot('final state')
  });
});
