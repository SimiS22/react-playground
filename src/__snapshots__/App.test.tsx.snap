// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test test 1: final state 1`] = `
<div>
  <div
    data-testid="myDiv"
  >
    final value
  </div>
  <button
    data-testid="myButton"
  >
    Change
  </button>
</div>
`;
